File: Python/QuickSort.py
def Swap(arr,a,b):
    temp=arr[a]
    arr[a]=arr[b]
    arr[b]=temp
    return


def Part(arr,start,end):
    pivot=arr[end]
    pindex=start

    for i in range(start,end):
        if(arr[i]<pivot):
            Swap(arr,i,pindex)
            pindex+=1


    temp=arr[pindex]
    arr[pindex]=arr[end]
    arr[end]=temp

    return pindex


def QuickSort(arr,start,end):
    if(start<end):
        pindex=Part(arr,start,end)
        if(pindex!=end):
            QuickSort(arr,start,pindex-1)
            QuickSort(arr,pindex+1,end)


    return


arr=(123,1,3456,2,3123,34,44,111)
L=list(arr)

QuickSort(L,0,7)
print(L)

File: Javascript/QuickSort.js
function partition(arr,start,end)
{
    var pivot = arr[end];
    var index = start;
    for(var i=start;i<end;i++)
    {
        if(arr[i]<pivot)
        {
            var temp = arr[i];
            arr[i] = arr[index];
            arr[index] = temp;
            index++;
        }
    }

    arr[end] = arr[index];
    arr[index] = pivot;
    return index;
}

function partitionHoare(arr,start,end)
{
    var i = start-1;
    var j = end+1;
    var pivot = arr[start];
    while(true)
    {
        do
        {
            i++;
        }while(arr[i]<pivot)

        do
        {
            j--;
        }while(arr[j]>pivot)

        if(i<j)
        {
            var temp = arr[i];
            arr[i] = arr[j];
            arr[j] =temp;
        }
        else
        {
            return j;
        }
    }
}
function sort(arr, start, end)
{
    if(start<end)
    {
        var pivot = partitionHoare(arr,start,end);
        sort(arr,start,pivot);
        sort(arr,pivot+1,end);
    }
}

var arr = [123,1,-123,34,-68,0,1234,-1,8,9,-1,-1,34,-68];
sort(arr,0,arr.length-1);
console.log(arr);